ebu_tt_live/bindings/__init__.py

```diff
@@ -1197,4 +1308,100 @@ def create_default_value(cls):
         )
         return instance
 
+    def _semantic_before_traversal(self, dataset, element_content=None, parent_binding=None):
+        self._semantic_register_id(dataset=dataset)
+
+
 raw.d_style_type._SetSupersedingClass(d_style_type)
+
+
+class d_body_type(SemanticValidationMixin, raw.d_body_type):
+
+    _attr_en_pre = {
+        (pyxb.namespace.ExpandedName(None, 'begin')).uriTuple(): BodyTimingValidationMixin._pre_timing_set_attribute,
+        (pyxb.namespace.ExpandedName(None, 'dur')).uriTuple(): BodyTimingValidationMixin._pre_timing_set_attribute,
+        (pyxb.namespace.ExpandedName(None, 'end')).uriTuple(): BodyTimingValidationMixin._pre_timing_set_attribute
```

Comment:
I'm not sure why these lines are present: none of these attributes is permitted on `body` in EBU-TT-D. Are they needed for something?